compiler/sigmatch.nim

```diff
@@ -1564,7 +1564,11 @@ proc matchesAux(c: PContext, n, nOrig: PNode,
           m.state = csNoMatch
           return
         m.baseTypeMatch = false
-        n.sons[a] = prepareOperand(c, formal.typ, n.sons[a])
+        if formal.typ.kind == tyVarargs and formal.typ.sons[0].kind == tyExpr and
+          m.calleeSym.magic notin SpecialSemMagics: # This is just as incorrect as before
```

Comment:
This is not helpful. "as incorrect as before"? what are these SpecialSemMagics that use varargs[expr]? Cannot we fix these instead?